tritium-caffeine/src/main/java/com/palantir/tritium/metrics/caffeine/CacheStats.java

```diff
@@ -0,0 +1,130 @@
+import java.util.Arrays;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.function.Supplier;
+import org.checkerframework.checker.index.qual.NonNegative;
+
+public final class CacheStats implements StatsCounter, Supplier<StatsCounter> {
+    private final String name;
+    private final Meter hitMeter;
+    private final Meter missMeter;
+    private final Meter loadSuccessMeter;
+    private final Meter loadFailureMeter;
+    private final Meter evictionsTotalMeter;
+    private final ImmutableMap<RemovalCause, Meter> evictionMeters;
+    private final LongAdder totalLoadNanos = new LongAdder();
+
+    /**
+     * Creates a {@link CacheStats} instance that registers metrics for Caffeine cache statistics.
+     * <p>
+     * Example usage for a {@link com.github.benmanes.caffeine.cache.Cache} or
+     * {@link com.github.benmanes.caffeine.cache.LoadingCache}:
+     * <pre>
+     *     LoadingCache&lt;Integer, String&gt; cache = Caffeine.newBuilder()
+     *             .recordStats(CacheStats.of(taggedMetricRegistry, "your-cache-name"))
+     *             .build(key -&gt; computeSomethingExpensive(key));
+     * </pre>
+     * @param taggedMetricRegistry tagged metric registry to add cache metrics
+     * @param name cache name
+     * @return Caffeine stats instance to register via
+     * {@link com.github.benmanes.caffeine.cache.Caffeine#recordStats(Supplier)}.
+     */
+    public static CacheStats of(TaggedMetricRegistry taggedMetricRegistry, @Safe String name) {
+        return new CacheStats(CacheMetrics.of(taggedMetricRegistry), name);
+    }
+
+    private CacheStats(CacheMetrics metrics, @Safe String name) {
+        this.name = name;
+        this.hitMeter = metrics.hit(name);
+        this.missMeter = metrics.miss(name);
+        this.loadSuccessMeter =
+                metrics.load().cache(name).result(Load_Result.SUCCESS).build();
+        this.loadFailureMeter =
+                metrics.load().cache(name).result(Load_Result.FAILURE).build();
+        this.evictionsTotalMeter = metrics.eviction(name);
+        this.evictionMeters = Arrays.stream(RemovalCause.values())
+                .collect(Maps.toImmutableEnumMap(cause -> cause, cause -> metrics.evictions()
+                        .cache(name)
+                        .cause(cause.toString())
+                        .build()));
+    }
+
+    @Override
+    public StatsCounter get() {
+        return this;
+    }
+
+    @Override
+    public void recordHits(@NonNegative int count) {
+        hitMeter.mark(count);
+    }
+
+    @Override
+    public void recordMisses(@NonNegative int count) {
+        missMeter.mark(count);
+    }
+
+    @Override
+    public void recordLoadSuccess(@NonNegative long loadTime) {
+        loadSuccessMeter.mark();
+        totalLoadNanos.add(loadTime);
+    }
+
+    @Override
+    public void recordLoadFailure(@NonNegative long loadTime) {
+        loadFailureMeter.mark();
+        totalLoadNanos.add(loadTime);
```

Comment:
I suppose we could make the meter into a timer, which also provides a “count” value in addition to percentiles, more timeseries though